MkDocs platform and diataxis framework

karenmajewicz opened the pull request:
Here is a PR that addresses #72 and.https://github.com/OpenGeoMetadata/metadata-issues/issues/46.

Preview here: https://karenmajewicz.github.io/opengeometadata.github.io/

I guess this is a fairly radical restructure, so I am hoping a few people can look it over. I updated the readme file to explain MkDocs and how to contribute. 

If we decide to go with this, we might need to update a few links on geoblacklight.org/docs that point to this site.  I'd also expect a little trial and error to make sure the custom domain works correctly.

rmseifried commented:
I like the way this is structured! 

Some observations / questions:
* The home page seems a little empty to me, but don't have ideas about what to change.
* On the home page, Schemas and Upgrading have to do with similar content - could they be combined, or at least placed next to each other?
* Is it possible to enable breadcrumbs on child pages (e.g. [References](https://karenmajewicz.github.io/opengeometadata.github.io/ogm-aardvark/references/))? Currently, if you navigate to one of those pages, the sidebar disappears and it isn't obvious how to return to the Aardvark table.
* I'd suggest moving [JSON Format](https://karenmajewicz.github.io/opengeometadata.github.io/JSON-format/) to "Explanation"
* [Configuring External Links](https://karenmajewicz.github.io/opengeometadata.github.io/configuring-external-links/) seems more like a "How-To Guide" to me. That said, we decided in the [current version](https://opengeometadata.org/docs/ogm-aardvark/references) to include this information together with the Aardvark field description. If we separate it from that content, there should at least be a link on the Aardvark field page pointing to this.

karenmajewicz commented:
Thanks for the review @rmseifried ! I am first looking into the breadcrumbs issue as that was something that prompted this to begin with and is not fixed.  

I think the problem is kind of the same as with our jekyll site. We don't directly include the Aardvark elements in any navigation, so it the site doesn't interpret them as breadcrumbs/subpages.

I published an experiment here: https://karenmajewicz.github.io/opengeometadata.github.io/ogm-aardvark/access-rights/
This adds each Aardvark page to the navigation.  However, we already have Aadvark documented in a grid and a table so maybe we would want to reconsider how it is organized....

karenmajewicz commented:
The latest commit includes several changes ([preview here](https://karenmajewicz.github.io/opengeometadata.github.io/)):
- adds breadcrumbs
- adds a homepage design
- moves the JSON Format page to Explanation
- splits the References into 2 pages (a list of URIs and a page explaining how to construct them) that are linked from the field documentation in the OGM Aardvark pages
- adds a page linking to the JSON Schema files
- increases font size


-----
Note: The nav can be organized with Section Headers, which may or may not have their own page. This affects which parts of the breadcrumb a user can click on. We can add pages for each section header, although it is a little faster to get to the content if they are omitted.  Currently, the Reference header has its own page, but the How-to Guides and the Explanation does not. The same goes for subheaders.

------
One thing I am wondering about is how the structure for Aardvark and for GBL 1.0 is a little different.

- With Aardvark, each element has its own page - these need to be added to the navigation and show up as nested pages in the left hand menu.
- With GBL 1.0, the elements are all documented on the same page and show up as items in the Table of Contents on the right hand menu.

Which method is easiest to navigate? By the way, each page has an option to turn on/off the sidebar nav or Table of Contents and I have experimented with both on various pages.

kgjenkins commented:
My first reaction is to prefer the current site layout, since I was just finally getting used to the new structure we established last year...

On pages like the [how-to guides](https://karenmajewicz.github.io/opengeometadata.github.io/share-on-ogm/) the sidebar links are getting cut off and so it's not clear what they go to:
![image](https://user-images.githubusercontent.com/3355358/226687593-010f3e9f-90ae-48db-84bf-a188cd1b60f0.png)

Also, the fixed-width font makes it a bit harder to read.

And I liked the indentation of 2nd-level items on the left of the current website.

These are all issues related to the appearance of the proposed site, not actually related to the organization of the content.  Could we either make the structural changes to the existing site, or else update the CSS display of the proposed site?

kgjenkins commented:
Regarding the GBL schemas, since we want to encourage use of Aardvark, I'd suggest grouping the GBL schemas together as legacy/deprecated, something like:
- Aardvark
- Deprecated schemas
    - GBL 1.0
    - GBL 0.5

karenmajewicz commented:
I made the changes Keith suggested: ([preview here](https://karenmajewicz.github.io/opengeometadata.github.io/)):

- put all navigation on the left hand side
- changed the font for readability and sizing
- shortened the text for headers to fit in the sidebar (this is  a best practice for good documentation anyways)
- increased the overall page width
------

In terms of which platform to use (MkDocs vs Jekyll), they both pretty much do the same thing: convert markdown to HTML. (I just find MkDocs easier to configure.)

karenmajewicz commented:
Interesting! On my browser, it didn't actually look like your screenshot before- I think I had previously chosen a nonstandard font that rendered as fixed width as a backup.

rmseifried commented:
This is looking really amazing! The comments I have at this point are hopefully fairly minor:

**Organization**
* Re: the different structure for Aardvark and GBL 1.0: with the new format that pulls section headings into the navigation panel navigation (which I love), I think the structure of GBL 1.0 is easier to navigate. Also, the great benefit of having all the fields on a single page is the ability to ctrl+f search terms. Dividing the fields into multiple pages (as we did with Aardvark) now seems rather inelegant - and is also more work to update. What is the rationale for the multi-page method?
* Pages related to `dct_references_s`: in the new format, the list of [Reference URIs](https://karenmajewicz.github.io/opengeometadata.github.io/reference-uris/) is now separate from the guide about [how to format the field](https://karenmajewicz.github.io/opengeometadata.github.io/configuring-external-links/), and these pages are in different sections of the website. To assist navigating between these sections, would it be possible to add buttons (or some other very obvious pointer) at the top of each page pointing to the other (in addition to the link back to the `dct_references_s` field information)?
* "Explanation" content organization: 
  * Could "About OGM Repositories" be moved to the top, in order to keep all the Aardvark content together?
  * Relatedly, using sub-headings in the navigation panel could be helpful here. Perhaps they could be: Repositories, Metadata, and a third category to hold "Further reading".
* "How-To Guides" content organization: similarly, we could consider using subheadings to divide the repository-related guide from the metadata guides. 
* "Reference" content organization: some of the pages seem like they could have more intuitive homes. Here's an idea about how to restructure:

```
- Schemas - move "Custom Fields"

- Crosswalks - ok

- Syntax
  - Field Suffixes (just the bottom part of the current "Solr Field Types" page) 
  - Custom Fields
  - Reference URIs

- Validation
  - JSON Schema Files

- Tools - ok

```
This would mean moving the tiny section on "Unique keys" currently on "Solr Field Types" elsewhere (perhaps it's more explanatory?). Also, "Solr Metadata Views" could be moved from "Syntax" to "How-To Guides > View Metadata in the Web" (or something similar), as it seems more instructive.



**Aesthetics**
* The Aardvark icon on the home page stands out as different from the others because it's wider / less round and more realistic / less schematic. What about [this cutie](https://thenounproject.com/icon/aardvark-931003/)?
* How-To Guides: I like the verb form "Share on OpenGeoMetadata" and would suggest using that for the other pages (e.g. "Upgrade to OGM Aardvark", "Create Metadata", etc.)
  * Relatedly, "External Links" --> "Configure External Links"

---
There may be other inter-page linking opportunities that we could add. For example, linking [Upgrade to Aardvark](https://karenmajewicz.github.io/opengeometadata.github.io/upgrading/) to the [Crosswalks](https://karenmajewicz.github.io/opengeometadata.github.io/aardvark-gbl-1-crosswalk/). We may want to consider a standard way of representing those inter-page links. Perhaps the `info` callout (as on the [Aardvark page](https://karenmajewicz.github.io/opengeometadata.github.io/ogm-aardvark/)?

karenmajewicz commented:
Here are some more changes from this last commit:

- used a different Aardvark icon that matches the GitHub icon better
- converted individual Aardvark element pages into a single page
- made groupings for each Tab split into repositories or metadata (or other groupings as needed)
- added more prominent button links between References & Reference URIs
- moved the small sections about Solr to explanation about Aardvark

kgjenkins commented:
Karen, this is looking really great!  The only thing I'd suggest changing is making the headings in the TOC stand out.  Adding this code to the extras.css should do the trick (keeping them black, instead of turning them grey):
```
.md-nav__item--section > .md-nav__link[for] {
    color: inherit;
}
```

I ended up diving into the word of building gh-pages from mkdocs just to test it, and so my branch also includes some code to do that.  You can see the result here: https://kgjenkins.github.io/opengeometadata.github.io/reference/

If you want to try the publishing via github actions, see here:
https://github.com/kgjenkins/opengeometadata.github.io/blob/main/.github/workflows/mkdocs.yml

rmseifried commented on docs/configure-external-links.md:
The link in the first line of text is broken - recommend using the same from the "Reference URI's" page:

```
This page provides guidance on how to format the [`dct_references_s`](../ogm-aardvark#references) field in either GBL 1.0 or OGM Aardvark.
```

rmseifried replied:
I think this is good to go live! I suggest one minor change and spotted a few other things that can be fixed in a future copyediting pull request, but for now, it would be great to get other folks' feedback on this reorg. Awesome job, @karenmajewicz !

karenmajewicz commented:
Thanks for the feedbaci - I will make the final changes and then work on getting it published.
